### PR TITLE
Documentation: Update documentation and examples to remove warnings due to Struct deprecation for Any

### PR DESCRIPTION
**Pull request by moderation**
*Description*: Update documentation and examples to remove warnings due to Struct deprecation for Any. I went to update the Google proxy example and fell into the rabbit hole of Struct deprecation in favor of Any. TODO is update templates in `/configs` which are currently generating deprecated config. Similarly the v1 to v2 scripts output deprecated config.
*Risk Level*: Low. Documentation or examples only. There are some YAML updates that are processed in testing but this should be OK.
*Testing*: `bazel test //test/...`
*Docs Changes*: included in PR
*Release Notes*: none required

### CODE REVIEW

**Comment by mattklein123** [comment]
Thanks a ton @moderation. Assigning over to @lizan. I do wonder if we should actually deprecate the struct version at all, and might consider undeprecating it and supporting both options for ease of configuration. cc @htuch

**Comment by moderation** [comment]
I think it is OK to deprecate Struct with the understanding it is going to be a lot of work to update all the references to the old way. I haven't even looked at the instances in C test code which I expect will be numerous.

The increase in complexity is primarily in defining hosts which used to be fairly concise. The new `load_assignment` method is verbose.

Lastly, there is a bunch of stuff like MySQL that haven't been changed over. The following config will still generate a deprecation warning but there isn't a way to use `typed_config` with MySQL. I haven't looked at Redis, Mongo etc.

```yaml
    filter_chains:
      - filters:
          - name: envoy.filters.network.mysql_proxy
            config:
              stat_prefix: mysql_stats
          - name: envoy.tcp_proxy
            typed_config:
              "@type": type.googleapis.com/envoy.config.filter.network.tcp_proxy.v2.TcpProxy
              stat_prefix: tcp_stats
              cluster: cluster_0
```

**Comment by lizan** [comment]
Thanks for taking this, my bad that didn't have a chance to make this happen earlier. 

@moderation the following for MySQL should work, no?

```yaml
    filter_chains:
      - filters:
          - name: envoy.filters.network.mysql_proxy
            typed_config:
              "@type": type.googleapis.com/envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy
              stat_prefix: mysql_stats
          - name: envoy.tcp_proxy
            typed_config:
              "@type": type.googleapis.com/envoy.config.filter.network.tcp_proxy.v2.TcpProxy
              stat_prefix: tcp_stats
              cluster: cluster_0
```

**Comment by htuch** [comment]
I think the migration should be pretty easy for most folks who want to keep Struct, they just put it in the Any field and it works. That said, I think the plan was to make this an extended deprecation cycles, e.g. a year or something.

**Comment by htuch** [comment]
FWIW, the main cost of keeping Struct that I see is API inconsistency. Going forward, we want to just use Any for opaque fields, so, turning down direct Struct fields seems the way to go (eventually).

**Comment by moderation** [comment]
@lizan I was able to work out your suggestion from the proto but when you try to run it you get:

```
[2019-02-21 11:02:38.169][2384][critical][main] [source/server/server.cc:86] error initializing configuration '/home/moderation/Library/envoyproxy/moderation/envoy/test/extensions/filters/network/mysql_proxy/mysql_test_config.yaml': Unable to parse JSON as proto (INVALID_ARGUMENT:(static_resources.listeners.filter_chains[0].filters[0].typed_config): invalid value Invalid type URL, unknown type: envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy for type Any): {"static_resources":{"listeners":{"filter_chains":[{"filters":[{"typed_config":{"stat_prefix":"mysql_stats","@type":"type.googleapis.com/envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy"},"name":"envoy.filters.network.mysql_proxy"},{"typed_config":{"@type":"type.googleapis.com/envoy.config.filter.network.tcp_proxy.v2.TcpProxy","cluster":"cluster_0","stat_prefix":"tcp_stats"},"name":"envoy.tcp_proxy"}]}],"name":"listener_0","address":{"socket_address":{"port_value":3307,"address":"0.0.0.0"}}},"clusters":{"load_assignment":{"endpoints":[{"lb_endpoints":[{"endpoint":{"address":{"socket_address":{"port_value":0,"address":"127.0.0.1"}}}}]}],"cluster_name":"cluster_0"},"name":"cluster_0","connect_timeout":"2s"}},"admin":{"address":{"socket_address":{"port_value":0,"address":"127.0.0.1"}},"access_log_path":"/dev/null"}}
```
Config used is:
```yaml
admin:
  access_log_path: /dev/null
  address:
    socket_address:
      address: 127.0.0.1
      port_value: 0
static_resources:
  clusters:
    name: cluster_0
    connect_timeout: 2s
    load_assignment:
      cluster_name: cluster_0
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: 127.0.0.1
                port_value: 0
  listeners:
    name: listener_0
    address:
      socket_address:
        address: 0.0.0.0
        port_value: 3307
    filter_chains:
      - filters:
          - name: envoy.filters.network.mysql_proxy
            typed_config:
              "@type": type.googleapis.com/envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy
              stat_prefix: mysql_stats
          - name: envoy.tcp_proxy
            typed_config:
              "@type": type.googleapis.com/envoy.config.filter.network.tcp_proxy.v2.TcpProxy
              stat_prefix: tcp_stats
              cluster: cluster_0
```
I suspect the proto at https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/filter/network/mysql_proxy/v1alpha1/mysql_proxy.proto doesn't have the required information to make this work.

**Comment by lizan** [comment]
@moderation it worked for me, which version of envoy are you using?  6c95ed9cbfa5e104208c951271817e487639574c worked for me

**Comment by moderation** [comment]
@lizan works for me too. I had the mysql extension compiled out. Doh. Once this PR is merged I'll take a stab at the outstanding items like the templates and tests.

**Comment by moderation** [comment]
PTAL @lizan

File: configs/original-dst-cluster/proxy_config.yaml [hunk 1]
```diff
@@ -7,24 +7,28 @@ static_resources:
...
-            codec_type: auto
-    use_original_dst: true
+          http_filters:
+          - name: envoy.router
+            typed_config: {}
+          codec_type: auto
+    listener_filters:
+    - name: envoy.listener.original_dst
+      typed_config: {}
+    name: original_dst
```
Comment (lizan) [inline configs/original-dst-cluster/proxy_config.yaml]: Is this the listener name, put it at top of the listener fields? While it is technically valid though I find it hard to read.

Comment (moderation) [reply configs/original-dst-cluster/proxy_config.yaml]: I removed the name field as it isn't mandatory. I added it during testing as this file needed a large refactor to get to v2 config.

**Pull request merged**